plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java

```diff
@@ -344,6 +345,7 @@ public HiveMetadata(
             String trinoVersion,
             HiveStatisticsProvider hiveStatisticsProvider,
             HiveRedirectionsProvider hiveRedirectionsProvider,
+            Optional<SystemTableProvider> additionalSystemTableProvider,
```

Comment:
Can we make it `Set<SystemTableProvider>` instead and return PARTITIONS/PROPERTIES also through a provider ?

Reply:
see https://github.com/trinodb/trino/pull/8476#discussion_r664485191